aviary/utils/doctape.py

```diff
@@ -435,3 +444,156 @@ def glue_keys(dict_of_dicts: dict, display=True) -> list:
     for key in all_keys:
         glue_variable(key, md_code=True, display=display)
     return all_keys
+
+
+def get_class_names(file_path) -> set:
+    """
+    Retrieve all class names from a given file and return as a set
+
+    Parameters
+    ----------
+    file_path: str or Path
+        file path
+    """
+    # Read the content of the file
+    with open(file_path, 'r') as file:
+        file_content = file.read()
+
+    # Parse the file content into an AST
+    tree = ast.parse(file_content)
+
+    # Extract class names
+    class_names = [
+        node.name for node in ast.walk(tree)
+        if isinstance(node, ast.ClassDef)
+    ]
+
+    return set(class_names)
+
+
+def get_function_names(file_path) -> set:
+    """
+    Get all function names in a given file and return as a set.
+
+    Parameters
+    ----------
+    file_path: str or Path
+        file path
+    """
+    # Read the content of the file
+    with open(file_path, 'r') as file:
+        file_content = file.read()
+
+    # Parse the file content into an AST
+    tree = ast.parse(file_content)
+
+    # Extract function names
+    function_names = [
+        node.name for node in ast.walk(tree)
+        if isinstance(node, ast.FunctionDef)
+    ]
+
+    return set(function_names)
+
+
+def glue_actions(cmd, curr_glued=None, glue_default=False, glue_choices=False, md_code=True):
+    """
+    Glue all Aviary CLI options
+
+    Parameters
+    ----------
+    cmd: str
+        Aviary command
+    curr_glued: list
+        the parameters that have been glued
+    glue_default: boolean
+        flag whether the default values should be glued.
+    """
+    parser = argparse.ArgumentParser()
+    _command_map[cmd][0](parser)
+    actions = [*parser._get_optional_actions(), *parser._get_positional_actions()]
+    for action in actions:
+        opt_list = action.option_strings
+        for opt in opt_list:
+            if opt not in curr_glued:
+                glue_variable(opt, md_code=md_code)
+                curr_glued.append(opt)
+        if action.dest not in curr_glued:
+            glue_variable(action.dest, md_code=md_code)
+            curr_glued.append(action.dest)
+        if glue_default:
+            if str(action.default) not in curr_glued:
+                glue_variable(str(action.default), md_code=True)
+                curr_glued.append(str(action.default))
```

Comment:
I think it would be good to track where the default value is coming from.
```suggestion
            action_default = f'{action.dest}_default')
            if action_default not in curr_glued:
                glue_variable(action_default, str(action.default), md_code=True)
                curr_glued.append(action_default )
```
otherwise, if we are just gluing the value (such as `False` for one and `True` for another) then if the default of an action changes to something that is also being glued elsewhere, the docs won't update

Reply:
You are right. I learned something new.